### PR TITLE
Changes in build for new service worker added

### PR DESCRIPTION
**Pull request by jpuri**
PR adds code to build the new service worker and minified file will be added in `dist` for all browsers except firefox.

The work is based on branch `mv3` to avoid merging to develop till MV3 related changes are completed and tested.

### CODE REVIEW

File: app/manifest/_base.json [hunk 1]
```diff
@@ -1,8 +1,7 @@
 {
   "author": "https://metamask.io",
   "background": {
-    "page": "background.html",
-    "persistent": true
+    "service_worker": "app-init.js"
   },
   "browser_action": {
     "default_icon": {
```
Comment (jpuri) [inline app/manifest/_base.json:7]: Manifest in this PR has minimal changes to just be able to run service worker, there is a separate task to do other changes in manifest.

**Comment by jpuri** [comment]
Build is broken in this PR, it should be fixed after a couple of following tasks are implemented.

The PR is not merged to develop, we plan to keep mv3 changes in separate branch.

File: app/scripts/app-init.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// eslint-disable-next-line
+self.oninstall = () => {
+  tryImport('./globalthis.js');
+  tryImport('./sentry-install.js');
+  tryImport('./runtime-lavamoat.js');
+  tryImport('./lockdown-more.js');
+  tryImport('./policy-load.js');
+
+  const fileList = [
+    /** FILE NAMES */
```
Comment (adonesky1) [inline app/scripts/app-init.js:22]: Do we not have some way to know the names of these files yet?

Comment (jpuri) [reply app/scripts/app-init.js:22]: These files are generated dynamically during build. They list currently is:

```
tryImport('./common-0.js');
  tryImport('./common-1.js');
  tryImport('./common-2.js');
  tryImport('./common-3.js');
  tryImport('./common-4.js');
  tryImport('./common-5.js');
  tryImport('./common-6.js');
  tryImport('./background-0.js');
  tryImport('./background-1.js');
  tryImport('./background-2.js');
  tryImport('./background-3.js');
  tryImport('./background-4.js');
  tryImport('./background-5.js');
  tryImport('./background-6.js');
  tryImport('./background-7.js');
```

Comment (adonesky1) [reply app/scripts/app-init.js:22]: Oh does this actually get replaced with the node writeFile method. Got it got it

File: development/build/scripts.js [hunk 3]
```diff
@@ -494,6 +535,19 @@ function createFactoredBuild({
...
+            bundleAppInitialiser({
+              jsBundles,
+              browserPlatforms,
+              buildType,
+              devMode,
+              ignoredFiles,
+              testing,
+              policyOnly,
+              shouldLintFenceFiles,
+            });
```
[the remark this answers is not in the document]
Comment (jpuri) [reply development/build/scripts.js:550]: This fn replaces file names in `app-init.js` and `babelify and minify` it: https://github.com/MetaMask/metamask-extension/pull/13975/files#diff-a05d1f077953d24c0d8ddecf64163da2653ba3b728e1ba1f18af6d770767b127R340

[the remark this answers is not in the document]
Comment (adonesky1) [reply development/build/scripts.js:550]: oh right of course.

**Review by adonesky1 — APPROVED** [review]

**Pull request merged**